src/invites/infrastructure/persistence/repositories/invite.repository.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InviteRepository } from '../invite.repository';
+import { InjectRepository } from '@nestjs/typeorm';
+import { InviteEntity } from '../entities/invite.entity';
+import { InviteMapper } from '../mappers/invite.mapper';
+import { FindOptionsWhere, In, Repository } from 'typeorm';
+import { Invite } from 'src/invites/domain/invite';
+import { User } from 'src/users/domain/user';
+import { IPaginationOptions } from 'src/utils/types/pagination-options';
+import { inviteStatusEnum } from 'src/inviteStatuses/invite-status.enum';
+import { EntityCondition } from 'src/utils/types/entity-condition.type';
+import { NullableType } from 'src/utils/types/nullable.type';
+
+@Injectable()
+export class InviteRelationalRepository implements InviteRepository {
+  constructor(
+    @InjectRepository(InviteEntity)
+    private readonly inviteRepository: Repository<InviteEntity>,
+  ) {}
+
+  async create(data: Invite): Promise<Invite> {
+    const persistenceModel = InviteMapper.toPersistence(data);
+    const newEntity = await this.inviteRepository.save(
+      this.inviteRepository.create(persistenceModel),
+    );
+    return InviteMapper.toDomain(newEntity);
+  }
+
+  async findOne(
+    fields: EntityCondition<Invite>,
+  ): Promise<NullableType<Invite>> {
+    const entity = await this.inviteRepository.findOne({
+      where: fields as FindOptionsWhere<InviteEntity>,
+      relations: ['workspace'],
+    });
+
+    return entity ? InviteMapper.toDomain(entity) : null;
+  }
+
+  async update(id: Invite['id'], payload: Partial<Invite>): Promise<Invite> {
+    const entity = await this.inviteRepository.findOne({
+      where: { id: id as number },
+      relations: ['workspace'],
+    });
+
+    if (!entity) {
+      throw new NotFoundException();
+    }
+
+    const updatedInvite = await this.inviteRepository.save(
+      this.inviteRepository.create(
+        InviteMapper.toPersistence({
+          ...InviteMapper.toDomain(entity),
+          ...payload,
+        }),
+      ),
+    );
+
+    return InviteMapper.toDomain(updatedInvite);
+  }
+
+  async findManyByEmails(emails: string[]): Promise<Invite[]> {
+    const entities = await this.inviteRepository.find({
+      where: {
+        invitee_email: In(emails),
+      },
+    });
+    return entities.map((entity) => InviteMapper.toDomain(entity));
+  }
```

Comment:
I'm unclear about the usage of this function.

Reply:
Unused function I will delete it.